lib/matrobjrowlist.gi

```diff
@@ -0,0 +1,83 @@
+#############################################################################
+##
+##  This file is part of GAP, a system for computational discrete algebra.
+##
+##  SPDX-License-Identifier: GPL-2.0-or-later
+##
+##  Copyright of GAP belongs to its developers, whose names are too numerous
+##  to list here. Please refer to the COPYRIGHT file for details.
+##
+
+
+
+############################################################################
+# Elementary matrix operations
+############################################################################
+
+############################################################################
+##
+#M  MultMatrixRow( <M>, <row>, <scalar> )
+##
+InstallMethod( MultMatrixRowLeft, "for a mutable IsRowListMatrix, one row number, and a scalar",
+  [ IsRowListMatrix and IsMutable, IsInt, IsObject ],
+  function( mat, row, scalar )
+  
+    mat[row] := scalar * mat[row];
+    
+  end );
+
+
+############################################################################
+##
+#M  MultMatrixRowRight( <M>, <row>, <scalar> )
+##
+InstallMethod( MultMatrixRowRight, "for a mutable IsRowListMatrix, one row number, and a scalar",
+  [ IsRowListMatrix and IsMutable, IsInt, IsObject ],
+  function( mat, row, scalar )
+  
+    mat[row] := mat[row] * scalar;
+    
+  end );
+
+
+############################################################################
+##
+#M  AddMatrixRows( <M>, <row1>, <row2>, <scalar> )
+##
+InstallMethod( AddMatrixRowsLeft, "for a mutable IsRowListMatrix, one row number, second row number, and a scalar",
```

Comment:
"one" and "second" together makes no sense; maybe "first" and "second"... but really, I'd write it as
```suggestion
InstallMethod( AddMatrixRowsLeft, "for a mutable IsRowListMatrix, two row numbers, and a scalar",
```

Reply:
Similar elsewhere

Reply:
Good point. Has been corrected.